stacks-signer/src/client/stacks_client.rs

```diff
@@ -188,6 +247,40 @@ impl StacksClient {
         Ok(Some(point))
     }
 
+    /// Helper function  that attempts to deserialize a clarity hext string as a list of signer slots and their associated number of signer slots
+    fn parse_signer_slots(&self, hex: &str) -> Result<Vec<(StacksAddress, u128)>, ClientError> {
+        debug!("Parsing signer slots: {hex}...");
+        // Due to .signers definition, the  signer slots is always a result of a list of tuples of signer addresses and the number of slots they have
+        // If this fails, we have bigger problems than the signer crashing...
+        let value = ClarityValue::try_deserialize_hex_untyped(hex)?
+            .expect_result()
+            .map_err(|e| ClientError::ReadOnlyFailure(e.to_string()))?;
```

Comment:
Instead of `ReadOnlyFailure`, do you think you could have an error type specific to Clarity deserialization failure?

Reply:
Can do!